modules/cfe_testcase/src/time_conversion_test.c

```diff
@@ -0,0 +1,150 @@
+**      limitations under the License.
+**
+** File: time_conversion_test.c
+**
+** Purpose:
+**   Functional test of basic Time Conversion APIs
+**
+*************************************************************************/
+
+/*
+ * Includes
+ */
+#include "cfe_test.h"
+
+
+void TestConvertMET2SCTime(void)
+{
+    UtPrintf("Testing: CFE_TIME_MET2SCTime");
+
+    /* std os time variables */
+    OS_time_t          start;
+    OS_time_t          end;
+    OS_time_t          difference;
+
+    /* Mission Elapsed Time */
+    CFE_TIME_SysTime_t MET;
+    /* MET + SCTF */
+    CFE_TIME_SysTime_t TAI;
+    /* MET + SCTF - Leap Seconds */
+    CFE_TIME_SysTime_t UTC;
+    /* Spacecraft Time */
+    CFE_TIME_SysTime_t SCTime;
+
+    /* Print buffers */
+    char timeBuf1[sizeof("yyyy-ddd-hh:mm:ss.xxxxx_")];
+    char timeBuf2[sizeof("yyyy-ddd-hh:mm:ss.xxxxx_")];
+
+    OS_GetLocalTime(&start);
+
+    /* Get Times */
+    MET = CFE_TIME_GetMET();
+    TAI = CFE_TIME_GetTAI();
+    UTC = CFE_TIME_GetUTC();
+
+    OS_GetLocalTime(&end);
+
+    /* calculate diffrence */
+    difference = OS_TimeSubtract(end, start);
+
+    /* 
+     * Convert - should produce a TAI or UTC at the moment of GetMET() 
+     *			 with a minimal time shift between 
+     */
+    SCTime = CFE_TIME_MET2SCTime(MET);
+
+    /* write Spacecraft Time into second buffer */
+    CFE_TIME_Print(timeBuf2, SCTime);
+
+    /* Check conversion */
+#if (CFE_MISSION_TIME_CFG_DEFAULT_TAI == true)
+    /* SCTime is TAI format */
+
+    /* avoid unused compiler warning */
+    (void)UTC;
+
+    /* write TAI into first buffer */
+    CFE_TIME_Print(timeBuf1, TAI);
+
+    /*
+     * CFE_TIME_Compare not possible because of the 
+     * temporal diffrence between get GetMET and GetTAI 
+     */
+    UtAssert_True(TimeInRange(TAI, SCTime, difference), "TAI (%s) = MET2SCTime (%s)", timeBuf1, timeBuf2);
+
+    /* this would be ok but it would repeat a whats done in CFE_TIME_MET2SCTime */
+    /* UtAssert_True(CFE_TIME_Compare(SCTime, CFE_TIME_Add(MET, STCF))); */
+#else
+    /* SCTime is UTC format */
+
+    /* avoid unused compiler warning */
+    (void)TAI;
+
+    /* write UTC into first buffer */
+    CFE_TIME_Print(timeBuf1, UTC);
+
+    /* 
+     * CFE_TIME_Compare not possible because of the
+     * temporal diffrence between get GetMETand GetUTC 
+     */
+    UtAssert_True(TimeInRange(UTC, SCTime, difference), "UTC (%s) = MET2SCTime (%s)", timeBuf1, timeBuf2);
+#endif
+}
+
+void TestConvertSubSeconds2MicroSeconds(void)
+{
+    UtPrintf("Testing: CFE_TIME_Sub2MicroSecs");
+
+    /* predefined amount of sub-seconds */
+    uint32 SUB = 400;
+    /* correct micro-seconds equal to the predefined sub-seconds */
+    uint32 MS = 93;
+    uint32 Sub2Micro;
+
+    /* run Sub2MicroSecs with predefined amount of sub-seconds and save result */
+    Sub2Micro = CFE_TIME_Sub2MicroSecs(SUB);
+
+    UtAssert_True(MS == Sub2Micro, "sub-seconds (%u) = Sub2MicroSecs (%u)", SUB, Sub2Micro);
```

Comment:
You should use the macro "UtAssert_UINT32_EQ(MS, Sub2Micro)" instead of UtAssert_True, it will handle the print out for you. Should also change the variable name of MS to something like ExpectedMS.

Reply:
Oh, right.